common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -363,22 +416,39 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
-                    break;
+                    return DeadlineRunStatus.TimedOut;
                 }
             }
 
-            task = pollTask();
+            task = pollTaskFrom(taskQueue);
             if (task == null) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 break;
             }
         }
+        lastExecutionTimeCallback.call(lastExecutionTime);
+        return DeadlineRunStatus.RanAtleastOnce;
+    }
 
-        this.lastExecutionTime = lastExecutionTime;
-        return true;
+    protected enum DeadlineRunStatus {
+        NoTasksFound,
+        TimedOut,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this constant name to match the regular expression '^([a-z][a-zA-Z0-9]*|[A-Z][A-Z0-9_v]*)$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00115)